apps/prairielearn/src/lib/editors.ts

```diff
@@ -879,9 +882,39 @@ export class QuestionAddEditor extends Editor {
 
     const fromPath = path.join(EXAMPLE_COURSE_PATH, 'questions', 'demo', 'calculation');
     const toPath = questionPath;
+
     debug(`Copy template\n from ${fromPath}\n to ${toPath}`);
     await fs.copy(fromPath, toPath, { overwrite: false, errorOnExist: true });
 
+    if (this.files != null) {
+      debug('Remove template files when file texts provided');
+      await fs.remove(path.join(toPath, 'question.html'));
+      await fs.remove(path.join(toPath, 'server.py'));
```

Comment:
This will unconditionally remove both `question.html` and `server.py` if any files are provided. Is this even needed, or will the `writeFile()` below replace these files anyway?

Reply:
Since we're copying from the basic question template, both its `question.html` and `server.py` are copied over. If a question only needs html, then that `server.py` contains unneeded code and can be removed. (I don't think there's a case where no html makes a valid question, so `question.html` should be assumed mandatory if `server.py` is provided)

Reply:
Good point. Does this apply to other files as well? I guess I'm bothered by the fact that at the moment we single out these two files for special treatment. Right now this is fine, because these are the only two we are generating in any case, but this function is being written as a general mechanism for updating questions.

I think that we should change the function to delete all existing files in the question if replacement files are provided, and document that this is what will happen. The only exception to this should be `info.json`, I guess? How is the updating of that file handled in this flow?

Reply:
Made a small update in b5c604cd1a9f33f625b695563a12dcab48f3d0f9- the new logic is that since the template has `question.html`, `server.py` and `info.json`, if _any_ files are provided, files _only_ exist if they are provided, _except_ for `info.json` which will default to the template file if not provided